packages/core/src/entity/EntityAssigner.ts

```diff
@@ -43,7 +43,11 @@ export class EntityAssigner {
       }
 
       if ([ReferenceType.MANY_TO_ONE, ReferenceType.ONE_TO_ONE].includes(props[prop]?.reference) && Utils.isDefined(value, true) && EntityAssigner.validateEM(em)) {
-        return EntityAssigner.assignReference<T>(entity, value, props[prop], em!, options);
+        if (options.updateNestedEntities && Utils.isEntity(entity[prop], false) && Utils.isPlainObject(value)) {
+          return EntityAssigner.assign(entity[prop], value, options);
+        }
+          return EntityAssigner.assignReference<T>(entity, value, props[prop], em!, options);
+
```

Comment:
bad ident, how come this was not cought by eslint 🤫 
```suggestion

        return EntityAssigner.assignReference<T>(entity, value, props[prop], em!, options);
```